src/modules/nft/index.js

```diff
@@ -0,0 +1,103 @@
+import Nft from './src';
+export default class NFT {
+  constructor({ network, address, web3 }) {
+    this.network = network;
+    this.address = address;
+    this.web3 = web3;
+    this.nft = new Nft({
+      network: this.network,
+      address: this.address,
+      setAvailableContracts: this.setAvailableContracts.bind(this),
+      web3: this.web3
+    });
+    this.availableContracts = [];
+    this.selectedContract = {};
+    this.currentActive = '';
+    this.currentPageState = '';
+    this.ready = false;
+  }
+
+  init(selectedContractOverride) {
+    return new Promise((resolve, reject) => {
+      try {
+        this.nft
+          .setup()
+          .then(selectedContract => {
+            if (selectedContractOverride) {
+              selectedContract = selectedContractOverride;
+            }
+            this.selectedContract = selectedContract;
+            return this.nft.getFirstTokenSet(selectedContract);
+          })
+          .then(currentActive => {
+            this.currentActive = currentActive;
+            this.currentPageState = this.currentActive.getPageState();
+            this.ready = true;
+            resolve(this.currentPageState);
+          })
+          .catch(reject)
+      } catch (e) {
+        reject(e);
+      }
+    });
+  }
+
+  get currentActiveContract() {
+    return this.currentActive.contract;
```

Comment:
this might error if this.currentActive is not set yet , maybe change this.currentActive = {} instead of '' ?